pitch.md

```diff
@@ -1,69 +1,172 @@
+I am also not interested with actually implementing a good type-checking system
+(i.e., one with good inference). The only important thing is that such
+a thing _could_ be made. Furthermore, how well it actually runs is far less important than what
+happens during compilation, _so much so_ that it would be fine if no actually runnable code
+could be made.
+
+Thus, the meat of the project is still in language design. The main question is how to
+give direct access to compile-time calculations. There are basically two ways
+to do this, and it might be possible to unify them. The first is directly manipulating
+abstract syntax trees. The second is a generic form of "property propagation", such
+as allows naive type-checking. That can be described as an operation on ASTs, though.
+So given a rudimentary underlying abstract syntax, which would probably not be any more
+complicated than a very simple subset of Haskell, the main question of the project
+becomes how to expressively manipulate and calculate the properties of ASTs.
 
 ### Scope
-_How big an idea is this? How ambitious is this project?_
 
+Extremely ambitious, if it were to be of practical use.
+It might be the case that the final product of a project like this would be
+no more than a language specification, and an outline of an implementation strategy,
+rather than something that actually ran. That would mean that nearly all the 
+work was in languages! But it would also be a rather unsatisfying conclusion.
+
+(Because of this, I have some reserve projects in mind.)
 
 ### Challenges and opportunities
-_Why is this a good project? What are some challenges you expect to face? How
-might you overcome them? What are some warning signs that the project has gotten
-off track, and how will you get the project back on track if needed?_
+
+The project can get off-track in several ways.
+
+The first is if it starts to turn into an infinite cycle of reading papers.
+If that is the case early-on, then I would say that I should abort the project altogether.
+Otherwise, it would just be something to avoid.
+
+The second is if things start to get implementation-heavy. Doing a project like this
+requires being OK with a well-defined but unimplemented result, because the details of
+implementation are probably just too much. So, in this case, I should just step back
+and accept a more theoretical result.
+
+Being on-track looks more like this. I find very rudimentary compile-time feature X.
+What is the most expressive way to describe how to _implement_ (not use) X? Does that match up
+well with how I want to describe the implementation of feature Y? Merge and iterate to more complex,
+but still pretty simple features (rudimentary dependent types, for example). Then, iterate downwards: What base features do I have that could be implemented using others?
+Reach a simple core that could be built up to implement the features I've explored. Then, see how far that can
+go towards implementing other things.
```

Comment:
Definitely like this idea of iterating downwards, it would be the most efficient way for you to try and stay focused but also continue getting more and more features. I bet your reading this week will help you figure out whether you think this is actually possible